network/src/peer_registry.rs

```diff
@@ -22,7 +22,7 @@ pub struct PeerRegistry {
     max_outbound: u32,
     // Only whitelist peers or allow all peers.
     whitelist_only: bool,
-    whitelist_peers: HashSet<PeerId>,
+    whitelist_peers: HashSet<Multiaddr>,
```

Comment:
The only usage of this struct is to find out whether an address is in the whitelist, so it is more appropriate to use a `HashSet<PeerId>` than hash set of address.